custom_components/tahoma/config_flow.py

```diff
@@ -34,13 +41,22 @@ async def validate_input(hass: core.HomeAssistant, data):
         _LOGGER.exception("Error when trying to log in to the TaHoma API")
         raise CannotConnect
 
+    return_dict = {"title": username}
+    controller.get_setup()
+    devices = controller.get_devices()
+    for key, device in devices.items():
+        if device.uiclass == TAHOMA_TYPE_HEATING_SYSTEM:
+            if TAHOMA_TYPE_HEATING_SYSTEM not in return_dict:
+                return_dict[TAHOMA_TYPE_HEATING_SYSTEM] = {}
+            return_dict[TAHOMA_TYPE_HEATING_SYSTEM][device.url] = device.label
+
     # If you cannot connect:
     # throw CannotConnect
     # If the authentication is wrong:
     # InvalidAuth
 
     # Return info that you want to store in the config entry.
-    return {"title": username}
+    return return_dict
```

Comment:
Why is this necessary? Won't this change the name of the Somfy TaHoma entry? Currently this is the username.

Reply:
I believe `return_dict = {"title": username}` line 44 ensures the behavior is the same as before in that regard. Am I wrong?

Reply:
I use the return from validate_input to populate the `user_input` with elements used by climate entities (if discovered) and the options flow